### ISSUE TITLE
My validation accuracy is always increasing or decreasing in steps of 0.2

### ISSUE THREAD

**Issue by muvazima** [stellargraph/stellargraph#1906]
Please choose an appropriate template from the "Template" drop-down above.
Hi, I was working on node classification using GraphSAGE. I have 65 nodes, 30 for training and 30 for validation. My validation accuracy is always 0.2,0.4,0.6 or 0.8 ever after tuning the hyperparameters. Is there any reason for this ?
```python
batch_size = 10
num_samples = [5, 2]
generator = GraphSAGENodeGenerator(Gs, batch_size, num_samples)
train_gen = generator.flow(train_subjects.index, train_targets, shuffle=True)
graphsage_model = GraphSAGE(
    layer_sizes=[32, 32], generator=generator, bias=True, dropout=0.5,
)
x_inp, x_out = graphsage_model.in_out_tensors()
prediction = layers.Dense(units=train_targets.shape[1], activation="softmax")(x_out)
model = Model(inputs=x_inp, outputs=prediction)

model.compile(optimizer=optimizers.Adam(lr=0.005),loss=losses.categorical_crossentropy,metrics=["acc"],)
```
```python
from tensorflow.keras.callbacks import EarlyStopping

es_callback = EarlyStopping(monitor="val_acc", patience=50, restore_best_weights=True)
test_gen = generator.flow(test_subjects.index, test_targets)
history = model.fit(
    train_gen, epochs=200, validation_data=test_gen, verbose=1, shuffle=False, class_weight=class_weights, callbacks=[es_callback]
)
```

![image](https://user-images.githubusercontent.com/61550134/116059578-9cc2fc80-a69e-11eb-8cd0-6b467b7855e8.png)
![image](https://user-images.githubusercontent.com/61550134/116059703-bc5a2500-a69e-11eb-899b-8a91fba6962f.png)

**Comment by huonw** [comment]
Thanks for getting in touch, and thanks for providing code in your issue. Highly quantised accuracy typically means a small set of labelled data. As you may've deduced, your plot stepping by 0.2 suggests that there's only 5 validation examples, meaning the only possible accuracies are multiples of 1/5 (0, 0.2, 0.4, ...).

Could you confirm the size/contents of `test_subjects` and `test_targets`, double checking that your splitting process has correctly got the 30 nodes and labels for them?

Given you mention 65 and 2 * 30, I wonder if you may've ended up with the 5 = 65 - 2 * 30 left-over nodes as your validation set?

**Comment by muvazima** [comment]
Thanks for your quick reply. That was indeed the issue!

**Comment by huonw** [comment]
Great!
